package/MDAnalysis/core/universe.py

```diff
@@ -1012,9 +1014,9 @@ def _fragdict(self):
         frags = tuple([AtomGroup(np.sort(ix), self) for ix in frag_indices])
 
         fragdict = {}
-        for f in frags:
+        for i, f in enumerate(frags):
             for a in f:
-                fragdict[a] = f
+                fragdict[a] = (i, f)
```

Comment:
if we do a `namedtuple` here, we can do `fragdict[self].ix` / `fragdict[self].fragment` later

Reply:
Cool! That sounds much more readable.